### ISSUE TITLE
Destorying modal is slow due to scrollbar reset

### ISSUE THREAD

**Issue by Remo** [bootstrap-vue/bootstrap-vue#1831]
I've got a component with quite a few nested components. It's some kind of a form builder and one form contains a lot of elements and each elements as well as each row has a few options. Those options I'm hiding by default by using a bootstrap-vue modal component.

That works well and reasonably fast, but when I unload the page, things get insanely slow.
The problem starts in `resetScrollbar` which is called a lot and eventually causes `querySelectorAll` to consume a lot of CPU time. You can see it in the profile below:

![image](https://user-images.githubusercontent.com/129864/40043297-5c9ead78-5824-11e8-96e2-f747b3dcf185.png)

Is there some way to get around that? Would it be possible to manually call that method if required and disable it by default? I know that I can customize the component and add a switch, but I'm looking for a solution that doesn't require me overriding code which won't get updated in the future. I'm also wondering if it wouldn't be enough to reset the scrollbar if we close one modal instead of doing so when it gets destroyed.

**Comment by emanuelmutschlechner** [comment]
The function `resetScrollbar` restores:

1. fixed content padding (`.fixed-top, .fixed-bottom, .is-fixed, .sticky-top`), 
1. sticky content margin (`.sticky-top`), and 
1. navbar toggler margin (`.navbar-toggler`)
1. body padding

What's the length of each node list for selectors 1 to 3?

**Comment by Remo** [comment]
It's 0 for all of them? My form is huge though, `document.querySelectorAll('.row').length` return `6285`

**Comment by rinick** [comment]
@Remo this is a performance issue in both the library and your implementation.  for things that has 6000 rows, it's better to use a virtual scrolling system, instead of creating all of them

**Comment by rinick** [comment]
@emanuelmutschlechner this problem is similar to the clickout issue, the scrollbar should be reset only when it's set in the first place, not always be called beforeDestroy.

**Comment by emanuelmutschlechner** [comment]
@rinick agree. this function should only be called if the modal is visible

**Comment by Remo** [comment]
@rinick I'm aware, I didn't think the customer would add that many elements. In reality he just added 198 rows, but there are a lot of nested elements. While it's not very nicely engineered, removing this one call increased the performance by several factors, that's why I thought something is wrong.

**Comment by rinick** [comment]
@Remo could you try the commit in my PR and see if it fixes your issue?

**Comment by Remo** [comment]
seems to improve things for me!
​

**Comment by rinick** [comment]
@Remo unfortunately my PR for this fix is still not approved yet
https://github.com/bootstrap-vue/bootstrap-vue/pull/1837
